src/whitenoise/accounts_groups.rs

```diff
@@ -0,0 +1,400 @@
+    pub async fn get_or_create(
+        whitenoise: &Whitenoise,
+        account_pubkey: &PublicKey,
+        mls_group_id: &GroupId,
+    ) -> Result<(Self, bool), WhitenoiseError> {
+        let (account_group, was_created) =
+            Self::find_or_create(account_pubkey, mls_group_id, &whitenoise.database).await?;
+        Ok((account_group, was_created))
+    }
+
+    /// Gets an AccountGroup for the given account and group, if it exists.
+    pub async fn get(
+        whitenoise: &Whitenoise,
+        account_pubkey: &PublicKey,
+        mls_group_id: &GroupId,
+    ) -> Result<Option<Self>, WhitenoiseError> {
+        let account_group =
+            Self::find_by_account_and_group(account_pubkey, mls_group_id, &whitenoise.database)
+                .await?;
+        Ok(account_group)
+    }
+
+    /// Gets all AccountGroups for the given account.
+    pub async fn all_for_account(
+        whitenoise: &Whitenoise,
+        account_pubkey: &PublicKey,
+    ) -> Result<Vec<Self>, WhitenoiseError> {
+        let groups = Self::find_all_for_account(account_pubkey, &whitenoise.database).await?;
+        Ok(groups)
+    }
+
+    /// Gets all visible AccountGroups for the given account.
+    /// Visible means: pending or accepted (not declined).
+    pub async fn visible_for_account(
+        whitenoise: &Whitenoise,
+        account_pubkey: &PublicKey,
+    ) -> Result<Vec<Self>, WhitenoiseError> {
+        let groups = Self::find_visible_for_account(account_pubkey, &whitenoise.database).await?;
+        Ok(groups)
+    }
+
+    /// Gets all pending AccountGroups for the given account.
+    pub async fn pending_for_account(
+        whitenoise: &Whitenoise,
+        account_pubkey: &PublicKey,
+    ) -> Result<Vec<Self>, WhitenoiseError> {
+        let groups = Self::find_pending_for_account(account_pubkey, &whitenoise.database).await?;
+        Ok(groups)
+    }
+
+    /// Accepts this group invite by setting user_confirmation to true.
+    pub async fn accept(&self, whitenoise: &Whitenoise) -> Result<Self, WhitenoiseError> {
+        let updated = self
+            .update_user_confirmation(Some(true), &whitenoise.database)
+            .await?;
+        Ok(updated)
+    }
+
+    /// Declines this group invite by setting user_confirmation to false.
+    /// The group will be hidden from the UI but remains in MLS.
+    pub async fn decline(&self, whitenoise: &Whitenoise) -> Result<Self, WhitenoiseError> {
+        let updated = self
+            .update_user_confirmation(Some(false), &whitenoise.database)
+            .await?;
+        Ok(updated)
+    }
+
+    /// Resets confirmation to pending state (None).
+    pub async fn reset_confirmation(
+        &self,
+        whitenoise: &Whitenoise,
+    ) -> Result<Self, WhitenoiseError> {
+        let updated = self
+            .update_user_confirmation(None, &whitenoise.database)
+            .await?;
+        Ok(updated)
+    }
```

Comment:
Do we actually want to support this? I'm not sure I understand how this would work tbh